### PR TITLE
Use docker-py and refactor code

### LINKED ISSUE #2: switch over to docker-py
https://github.com/docker/docker-py would make life much easier

### PR DESCRIPTION
**Pull request by kaczmarj**
Resolves #2

### CODE REVIEW

**Comment by kaczmarj** [comment]
Most recent push adds tests for Dockerfile 

addresses #5

File: src/docker/docker_image.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    Returns
+    -------
+    volumes : dict
+        Dictionary of volumes to be passed to `volumes` kwargs of
+        `docker.api.build()`.
+    """
+    volumes = {}
+    for this_volume in volume_mapping:
+        try:
```
Comment (mgxd) [inline src/docker/docker_image.py:36]: maybe instead of try we can check len of tuple

File: src/docker/docker_image.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    events = client.events(decode=True)
+    event_logger = EventLogger(events, name="Event-Logger", daemon=True)
+    event_logger.start()
+
+    def __init__(self, path=None, fileobj=None, **kwargs):
+        self.path = path
+        self.fileobj = fileobj
+        # kwargs?
+
+        # Build image. This method uses the higher-level Docker API. To get the
```
Comment (mgxd) [inline src/docker/docker_image.py:102]: this is fine for now - we can add command line stream in the future

File: src/docker/dockerfile.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    ----------
+    specs : dict
+        Software specifications for the Dockerfile.
+    savedir : str
+        Directory in which to save Dockerfile (and optionally Conda environment
+        YAML file).
+    deps_method : {'apt-get', 'yum'}
+        Method to use to get dependencies.
+    """
+    def __init__(self, specs, savedir, deps_method):
```
Comment (mgxd) [inline src/docker/dockerfile.py:32]: should this accept kwargs?

File: src/docker/fsl.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+"""Class to add FSL installation to Dockerfile."""
+from __future__ import absolute_import, division, print_function
+
+from .utils import indent
+from ..utils import logger
+
+
+INSTALL_FSL = {
```
Comment (mgxd) [inline src/docker/fsl.py:8]: might want to store this separately as it grows

**Review by mgxd — APPROVED** [review]
LGTM!

**Pull request merged**